regions/io/fits/tests/test_fits_region.py

```diff
@@ -0,0 +1,114 @@
+
+import distutils.version as vers
+import pytest
+
+from numpy.testing import assert_allclose
+import astropy.version as astrov
+from astropy.utils.data import get_pkg_data_filename
+from astropy.table import Table
+from astropy.coordinates import SkyCoord
+from astropy import units as u
+
+from ..read import FITSRegionParser
+from ..write import fits_region_objects_to_table
+from ..core import FITSRegionParserError
+
+from ....shapes.circle import CircleSkyRegion
+
+_ASTROPY_MINVERSION = vers.LooseVersion('1.1')
+_ASTROPY_VERSION = vers.LooseVersion(astrov.version)
+
+implemented_region_types = ('ellipse', 'circle', 'box', 'polygon', 'point',
+                            'annulus', 'elliptannulus')
+
+
+@pytest.mark.parametrize('filename', ['data/region.fits'])
+def test_file_fits(filename):
+
+    filename = get_pkg_data_filename(filename)
+    table = Table.read(filename)
+    table['X'] = table['X'].astype(list)
+    table['Y'] = table['Y'].astype(list)
+    table.add_row([[1, 2, 3, 4], [5, 6, 7, 8], 'polygon', 0, 0, 8])
+
+    shapes = FITSRegionParser(table, 'warn').shapes
+
+    assert shapes[0].region_type == 'circle'
+    assert shapes[1].region_type == 'rectangle'
+    assert shapes[2].region_type == 'ellipse'
+    assert shapes[3].region_type == 'rectangle'
+    assert shapes[4].region_type == 'circleannulus'
+    assert shapes[5].region_type == 'point'
+    assert shapes[6].region_type == 'point'
+    assert shapes[7].region_type == 'polygon'
```

Comment:
There are 7 rows in the FITS table and shapes[7] gives me an IndexError

Reply:
The FITS table contains only seven rows. I added the 8th row to the table after reading the fits sample to cover the case for `polygon`